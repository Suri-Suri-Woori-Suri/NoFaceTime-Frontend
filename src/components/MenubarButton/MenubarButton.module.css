.MenuButton {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 80px;
  height: 80px;
  border: 1px solid grey;
  border-radius: 5%;
}

.MenuTitle {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-family: Helvetica;
  font-weight: 800;
  width: 100%;
  font-size: 10px;
  padding: 0;
  margin-bottom: -10px;
}

i {
  font-size: 36px;
}
