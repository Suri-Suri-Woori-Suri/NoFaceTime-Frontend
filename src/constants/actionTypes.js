export const GET_USER_DATA_FROM_DB = "GET_USER_DATA_FROM_DB";
export const ADD_ROOM = "ADD_ROOM";
export const DELETE_ROOM = "DELETE_ROOM";
export const ADD_GROUP = "ADD_GROUP";
export const DELETE_GROUPS = "DELETE_GROUPS";
export const ADD_MEMBERS = "ADD_MEMBERS";
export const DELETE_MEMBERS = "DELETE_MEMBERS";
