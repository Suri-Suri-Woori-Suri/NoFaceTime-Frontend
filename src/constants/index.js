export const API_METHOD = {
  GET: 'GET',
  POST: 'POST',
  PATCH: 'PATCH',
  DELETE: 'DELETE'
};

export const MENU_MODE = {
  OUT: 'Out',
  MIC: 'mic',
  MUTE: 'mute',
  INVITE: 'Invite',
  STUDENTS: 'Students',
  PUBLIC_CHAT: 'PublicChat',
  SCREEN_SHARE: 'ScreenShare',
  QUESTION_CHAT: 'QuestionChat'
};

export const MY_VIDEO_MODE = {
  HOST: 'host',
  PEER: 'peer'
};

export const FACE_STATUS = {
  DEFAULT: 'default',
  NEUTRAL: 'neutral',
  HAPPY: 'happy',
  SAD: 'sad',
  ANGRY: 'angry',
  DISGUSTED: 'disgusted',
  FEARFUL: 'fearful',
  SURPRIZED: 'surprized'
};

export const SOCKET_EVENT = {
  JOIN_ROOM: 'join room',
  JOIN_MYSELF: 'join myself',
  JOIN_NEW_MEMBER: 'join new member',
  SEND_SIGNAL_FROM_NEW_MEMBER: 'send signal from new member',
  CONVEY_SIGNAL_FROM_NEW_MEMBER: 'convey signal from new member',
  SENDBACK_SIGANL_FROM_EXISTING_MEMBER: 'sendback signal from existing member',
  CONVEY_SIGNAL_FROM_EXISTING_MEMBER: 'convey signal from existing member',
  MESSAGE_PUBLIC: 'message public',
  MESSAGE_SECRET: 'message secret',
  DUPLICATE: 'duplicate',
  LEAVE_ROOM: 'leave room',
  MEMBER_LEFT: 'member left',
  DISCONNECT: 'disconnect',
};

export const BACKGROUND_IMG = {
  NOTEBOOK: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/notebook.jpg',
  DETECTING_FACE: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/detecting-face.gif',
  WAITING_HOST: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/waiting-host.png',
  WAITING_MEMBER: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/waiting-member.png',
  LOADING: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/loading.png',
  RIGHTSIDE: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/background-image/rightside.png'
};

export const EMOJI = {
  NEUTRAL: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/smile.png',
  HAPPY: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/happy.png',
  SAD: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/sad.png',
  ANGRY: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/angry.png',
  ANGRY_WITH_WORD: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/angry-with-word.png',
  DISGUSTED: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/disgusted.png',
  FEARFUL: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/fearful.png',
  SURPRIZED: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/surprized.png',
  NO_FACE: 'https://two-face-time.s3.ap-northeast-2.amazonaws.com/emoji/no-face.png'
};
