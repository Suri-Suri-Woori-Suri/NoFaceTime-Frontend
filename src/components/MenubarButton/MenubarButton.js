import React from 'react';
import { MENU_MODE } from '../../constants/index';
import styles from './MenubarButton.module.css';

const MenubarButton = ({ handleClick, stopVideo }) => {
  const {
    NOTE,
    INVITE,
    STUDENTS,
    PUBLIC_CHAT,
    SCREEN_SHARE,
    QUESTION_CHAT } = MENU_MODE;

  const menuIcons = [
    [PUBLIC_CHAT, "fas fa-user-friends"],
    [QUESTION_CHAT, "fas fa-chalkboard-teacher"],
    [SCREEN_SHARE, "fas fa-desktop"],
    [NOTE, "fas fa-book-open"],
    [STUDENTS, "far fa-laugh"],
    [INVITE, "fas fa-user-plus"],
  ];

  const goBack = () => {
    stopVideo();
    window.history.back();
  }

  const menus = menuIcons.map((icon, index) => {
    const [menuName, iconName] = icon;
    return (
      <button
        key={index}
        className={styles.MenuButton}
        name={menuName}
        onClick={(e) => handleClick(e)}>
        <i className={iconName}></i>
        <p className={styles.MenuTitle}>{menuName}</p>
      </button>
    );
  })

  return (
    <>
    {menus}
    <button onClick={goBack}>Leave</button>
    </>
  );
};

export default MenubarButton;
