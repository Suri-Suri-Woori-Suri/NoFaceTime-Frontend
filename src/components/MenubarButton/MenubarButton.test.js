import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import MenubarButton from './MenubarButton';

import { MENU_MODE } from '../../constants/index';

describe('<MenubarButton />', () => {
  afterEach(cleanup);

  const {
    OUT,
    INVITE,
    STUDENTS,
    PUBLIC_CHAT,
    SCREEN_SHARE,
    QUESTION_CHAT } = MENU_MODE;
  const mockOnClick = jest.fn();
  const wrapper = mount(<MenubarButton handleClick={mockOnClick} />);

  it('should be invoked when button is clicked', () => {
    render(<MenubarButton handleClick={mockOnClick} />);

    fireEvent.click(screen.getByText(OUT));
    fireEvent.click(screen.getByText(INVITE));
    fireEvent.click(screen.getByText(STUDENTS));
    fireEvent.click(screen.getByText(PUBLIC_CHAT));
    fireEvent.click(screen.getByText(SCREEN_SHARE));
    fireEvent.click(screen.getByText(QUESTION_CHAT));

    expect(mockOnClick).toHaveBeenCalledTimes(6);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
