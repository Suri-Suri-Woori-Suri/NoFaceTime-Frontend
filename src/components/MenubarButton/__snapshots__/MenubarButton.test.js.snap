// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenubarButton /> matches snapshot 1`] = `
<MenubarButton
  handleClick={[MockFunction]}
>
  <button
    className="MenuButton"
    key="0"
    name="PublicChat"
    onClick={[Function]}
  >
    <i
      className="fas fa-user-friends"
    />
    <p
      className="MenuTitle"
    >
      PublicChat
    </p>
  </button>
  <button
    className="MenuButton"
    key="1"
    name="QuestionChat"
    onClick={[Function]}
  >
    <i
      className="fas fa-chalkboard-teacher"
    />
    <p
      className="MenuTitle"
    >
      QuestionChat
    </p>
  </button>
  <button
    className="MenuButton"
    key="2"
    name="ScreenShare"
    onClick={[Function]}
  >
    <i
      className="fas fa-desktop"
    />
    <p
      className="MenuTitle"
    >
      ScreenShare
    </p>
  </button>
  <button
    className="MenuButton"
    key="3"
    name="Students"
    onClick={[Function]}
  >
    <i
      className="far fa-laugh"
    />
    <p
      className="MenuTitle"
    >
      Students
    </p>
  </button>
  <button
    className="MenuButton"
    key="4"
    name="Invite"
    onClick={[Function]}
  >
    <i
      className="fas fa-user-plus"
    />
    <p
      className="MenuTitle"
    >
      Invite
    </p>
  </button>
  <button
    className="MenuButton"
    key="5"
    name="Out"
    onClick={[Function]}
  >
    <i
      className="fas fa-sign-out-alt"
    />
    <p
      className="MenuTitle"
    >
      Out
    </p>
  </button>
</MenubarButton>
`;
